Vagrantfile

```diff
@@ -0,0 +1,85 @@
+Vagrant.configure("2") do |config|
+  private_ip = {
+    "minio" => "10.0.11.10",
+    "postgres" => "10.0.11.11",
+    "redis" => "10.0.11.12"
+  }
+
+  config.vm.define "minio" do |minio|
+    env = {
+      "MINIO_ACCESS_KEY" => "development",
+      "MINIO_SECRET_KEY" => "development"
+    }
+    minio.vm.synced_folder ".", "/vagrant", disabled: true
+    minio.vm.network "forwarded_port", guest: 9000, host: 9000
+    minio.vm.network "private_network", ip: private_ip["minio"]
+
+    minio.vm.provider "docker" do |docker, override|
+      docker.image = "minio/minio"
+      docker.cmd = ["minio", "server", "/data"]
+      docker.env = env
+      docker.create_args = ["-v", "minio_data:/data"]
+    end
+
+    minio.vm.provider "virtualbox" do |virtualbox, override|
+      override.vm.box = "elegoev/ubuntu-18.04-minio"
+      virtualbox.memory = 1024
+      virtualbox.cpus = 2
```

Comment:
these might be excessively high core counts for my machine. Maybe we can default to 1 core for minio, redis and postgres?